db.py

```diff
@@ -80,20 +92,25 @@ def get_by_user(subclass, user):
 	@classmethod
 	def get_by_code(subclass, code):
 		try:
-			return subclass.select().where(subclass.code == code).get()
+			return subclass.select().where(subclass.code == str(code)).get()
 		except DoesNotExist:
 			return None
 
 	@classmethod
 	def upsert(subclass, **kwargs):
-		# FIXME temporary sanity-preserving patch until we get to issue #64.
-		if not isinstance(kwargs.get('code', None), str):
-			raise Exception('Need a code string, not an object')
 		return subclass.insert(kwargs).on_conflict_replace().execute()
 
 class PendingEmail(CodePivot):
 	email = TextField(null=False)
 
+	# This is all for automatic conversion to strings.
+	def __str__(self):
```

Comment:
Not a fan of this because it violates the principle of least surprise. A PendingEmail contains the user the email is for, the code used to confirm it, and the email itself. I would not expect using it in a string context to give me the code, and I also would not expect this to support addition at all.

Reply:
Removed it.